onnxruntime/core/optimizer/graph_optimizer_registry.cc

```diff
@@ -10,8 +10,34 @@ using namespace onnxruntime;
 using namespace ::onnxruntime::common;
 
 namespace onnxruntime {
+GraphOptimizerRegistry::GraphOptimizerRegistry(const onnxruntime::SessionOptions* sess_options,
+                                               const onnxruntime::IExecutionProvider* cpu_ep,
+                                               const logging::Logger* logger) : session_options_(sess_options),
+                                                                                cpu_ep_(cpu_ep),
+                                                                                logger_(logger) {}
 
-GraphOptimizerRegistry::GraphOptimizerRegistry() {}
+common::Status GraphOptimizerRegistry::CreatePredefinedSelectionFuncs() {
+  transformer_name_to_selection_func_[kConstantFoldingDQ] = ConstantFoldingDQFuncs::Select;
+
+  return Status::OK();
+}
+
+// Create and initialize the graph optimizer registry instance as a singleton.
+const GraphOptimizerRegistry& GraphOptimizerRegistry::Create(const onnxruntime::SessionOptions* sess_options,
+                                                             const onnxruntime::IExecutionProvider* cpu_ep,
+                                                             const logging::Logger* logger) {
+  if (!graph_optimizer_registry_) {  // First Check (without locking)
+    std::lock_guard<std::mutex> lock(registry_mutex_);
+    if (!graph_optimizer_registry_) {  // Second Check (with locking)
+      graph_optimizer_registry_ = std::unique_ptr<GraphOptimizerRegistry>(new GraphOptimizerRegistry(sess_options, cpu_ep, logger));
+      graph_optimizer_registry_->CreatePredefinedSelectionFuncs();
+      return *graph_optimizer_registry_;
+    }
+  }
+
+  LOGS(*graph_optimizer_registry_->GetLogger(), WARNING) << "The GraphOptimizerRegistry instance has been created before. Simply returning the existing instance.";
```

Comment:
nit: I'd probably put this at INFO level unless we think something bad could be happening like completely different SessionOptions.